templates/cluster-template-internal.yaml

```diff
@@ -117,80 +114,93 @@ spec:
       bootstrap:
         configRef:
           apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
           kind: KubeadmConfigTemplate
-          name: ${CLUSTER_NAME}-md
+          name: "${CLUSTER_NAME}-md-0"
       infrastructureRef:
         apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
         kind: VirtinkMachineTemplate
-        name: ${CLUSTER_NAME}-md
+        name: "${CLUSTER_NAME}-md-0"
 ---
 apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
 kind: KubeadmConfigTemplate
 metadata:
-  name: ${CLUSTER_NAME}-md
+  name: "${CLUSTER_NAME}-md-0"
 spec:
   template:
     spec:
       joinConfiguration:
         nodeRegistration:
           kubeletExtraArgs:
-            provider-id: virtink://{{ ds.meta_data.instance_id }}
+            provider-id: "virtink://{{ ds.meta_data.instance_id }}"
           ignorePreflightErrors:
             - SystemVerification
 ---
 apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
 kind: VirtinkMachineTemplate
 metadata:
-  name: ${CLUSTER_NAME}-md
+  name: "${CLUSTER_NAME}-md-0"
 spec:
   template:
     spec:
       vmSpec:
+        affinity:
+          podAntiAffinity:
+            preferredDuringSchedulingIgnoredDuringExecution:
+              - weight: 100
+                podAffinityTerm:
+                  topologyKey: kubernetes.io/hostname
+                  labelSelector:
+                    matchExpressions:
+                      - key: cluster.x-k8s.io/deployment-name
+                        operator: In
+                        values:
+                          - "${CLUSTER_NAME}-md-0"
         runPolicy: Once
         instance:
           cpu:
-            sockets: ${VIRTINK_MACHINE_CPU_COUNT}
-            coresPerSocket: 1
+            sockets: 1
+            coresPerSocket: ${VIRTINK_WORKER_MACHINE_CPU_CORES:=2}
           memory:
-            size: ${VIRTINK_MACHINE_MEMORY_SIZE}
+            size: ${VIRTINK_WORKER_MACHINE_MEMORY_SIZE:=4Gi}
           kernel:
-            image: ${VIRTINK_MACHINE_KERNEL_IMAGE}
-            cmdline: console=ttyS0 root=/dev/vda rw
+            image: "${VIRTINK_WORKER_MACHINE_KERNEL_IMAGE:=smartxworks/capch-kernel-5.15.12}"
+            cmdline: "console=ttyS0 root=/dev/vda rw"
           disks:
             - name: rootfs
           interfaces:
             - name: pod
         volumes:
           - name: rootfs
             containerRootfs:
-              image: ${VIRTINK_MACHINE_ROOTFS_IMAGE}
-              size: ${VIRTINK_MACHINE_ROOTFS_SIZE}
+              image: "${VIRTINK_WORKER_MACHINE_ROOTFS_IMAGE:=smartxworks/capch-rootfs-1.24.0}"
+              size: ${VIRTINK_WORKER_MACHINE_ROOTFS_SIZE:=4Gi}
         networks:
           - name: pod
             pod: {}
 ---
 apiVersion: cluster.x-k8s.io/v1beta1
 kind: MachineHealthCheck
 metadata:
-  name: ${CLUSTER_NAME}-worker-unhealthy-5m
+  name: "${CLUSTER_NAME}-md-0-unhealthy-5m"
 spec:
-  clusterName: ${CLUSTER_NAME}
+  clusterName: "${CLUSTER_NAME}"
+  maxUnhealthy: 75%
```

Comment:
The maxUnhealthy percentage should be 100% (default) ?

Reply:
We allow all worker nodes to be recreated instead of 75% of them, remediation won't be allowed when set percentage to 75%.

Right, without CNI be installed, all nodes will be in NotReady state. The MHC controller will try to recreate the "failed" one in 5min, but I think it doesn't matter.